File: src/main/java/com/github/sgtsilvio/gradle/android/retrofix/backport/Backport.java
package com.github.sgtsilvio.gradle.android.retrofix.backport;

import com.github.sgtsilvio.gradle.android.retrofix.transform.MethodMap;
import com.github.sgtsilvio.gradle.android.retrofix.transform.TypeMap;
import javassist.ClassPool;
import org.jetbrains.annotations.NotNull;

/**
 * @author Silvio Giebl
 */
public interface Backport {

    void apply(@NotNull ClassPool classPool, @NotNull TypeMap typeMap, @NotNull MethodMap methodMap);
}

File: src/main/kotlin/com/github/sgtsilvio/gradle/android/retrofix/backport/Backport.kt
package com.github.sgtsilvio.gradle.android.retrofix.backport

import com.github.sgtsilvio.gradle.android.retrofix.transform.ClassMap
import com.github.sgtsilvio.gradle.android.retrofix.transform.MethodMap

/**
 * @author Silvio Giebl
 */
interface Backport {

    val indicatorClass: String

    fun isInstrumentable(className: String): Boolean

    fun apply(classMap: ClassMap, methodMap: MethodMap)
}
